test/Table_Tests/src/Table_Spec.enso

```diff
@@ -891,6 +895,68 @@ spec =
+            t.filter "B" (Filter_Condition.Is_In [True, Nothing]) . at "B" . to_vector . should_equal []
+            t.filter "C" (Filter_Condition.Is_In [True, Nothing]) . at "C" . to_vector . should_equal [True, Nothing]
+            t.filter "A" (Filter_Condition.Is_In [False]) . at "A" . to_vector . should_equal [False]
+            t.filter "B" (Filter_Condition.Is_In [False]) . at "B" . to_vector . should_equal [False, False, False]
+            t.filter "C" (Filter_Condition.Is_In [False, False]) . at "C" . to_vector . should_equal [False]
+
+            t2 = Table.new [["ints", [1, 2, 3]], ["doubles", [1.2, 0.0, 1.0]]]
+            t2.filter "ints" (Filter_Condition.Is_In [2.0, 1.5, 3, 4]) . at "ints" . to_vector . should_equal [2, 3]
+            t2.filter "doubles" (Filter_Condition.Is_In [0.1, 1, 3, 1.2]) . at "doubles" . to_vector . should_equal [1.2, 1.0]
+
+        Test.specify "should perform `Is_In` efficiently for builtin types" <|
+            first_day = Date_Time.new 2000 1 1
+            make_date x = first_day + x.seconds
+            init = Duration.time_execution <|
+                t = Table.new [["X", (200.up_to 10000 . to_vector)]]
+                vec = 4000.up_to 13000 . to_vector
+                expected_vector = 4000.up_to 10000 . to_vector
+                expected_vector_2 = 200.up_to 10000 . with_step 2 . to_vector
+                dates_vec = vec.map make_date
+                bool_vec = Vector.fill 7000 True
+                date_col = t.at "X" . map make_date
+                [t, vec, expected_vector, expected_vector_2, dates_vec, bool_vec, date_col]
+            t = init.second . at 0
+            vec = init.second . at 1
+            expected_vector = init.second . at 2
+            expected_vector_2 = init.second . at 3
+            dates_vec = init.second . at 4
+            bool_vec = init.second . at 5
+            date_col = init.second . at 6
+
+            expected_max_time_ms = init.first.total_milliseconds * 2
+            check_timing name ~action =
+                res = Duration.time_execution action
+                runtime_ms = res.first.total_milliseconds
+                if runtime_ms > expected_max_time_ms then
+                    Test.fail "Expected `Is_In` on "+name+" to be efficient, but it took "+runtime_ms.to_text+"ms while initialization itself took just "+expected_max_time_ms.to_text+"ms."
+
+            check_timing "integers" <|
+                t.filter "X" (Filter_Condition.Is_In vec) . at "X" . to_vector . should_equal expected_vector
+
+            check_timing "booleans" <|
+                t.filter (t.at "X" % 2 == 0) (Filter_Condition.Is_In bool_vec) . at "X" . to_vector . should_equal expected_vector_2
+
+            check_timing "dates" <|
+                t.filter date_col (Filter_Condition.Is_In dates_vec) . at "X" . to_vector . should_equal expected_vector
```

Comment:
I did a timing check just to have a decent comparison and:
```
init = 501
integers = 31
booleans = 28
dates = 35
```

so `init*2` seems to still have a decent buffer.


For example if I disable the vectorized variant for booleans it gives:
```
    - [FAILED] should perform `Is_In` efficiently for builtin types [14626ms]
        Reason: Expected `Is_In` on booleans to be efficient, but it took 13852ms while initialization itself took just 1472ms.
```

So we can see that 2*init is a decent buffer - for good implementation it gives enough time, but the wrong implementation exceeds that greatly anyway. Let's see how it behaves on the CI though.